### PR TITLE
refactor(rust): `NodeManager` constructor 

### LINKED ISSUE #3527: Refactor `NodeManager` constructor
Currently `NodeManager::create` function takes 10 arguments. It's time to move those arguments into into one or few structures

https://github.com/build-trust/ockam/blob/92943069471cd6eface5bf0e64be5531394ded09/implementations/rust/ockam/ockam_api/src/nodes/service.rs#L136-L151

---

_We love helping new contributors!_
_If you have questions or need help as you work on your first Ockam contribution, please leave a comment on [this discussion](https://github.com/ockam-network/ockam/discussions/1081)._
_If you're looking for other issues to contribute to, checkout [this discussion](https://github.com/build-trust/ockam/discussions/3206) and labels -  https://github.com/build-trust/ockam/labels/good%20first%20issue or https://github.com/build-trust/ockam/labels/help%20wanted_

### PR DESCRIPTION
**Pull request by vncsalencar**
## Current Behavior
Current version of `NodeManager::create` receives 11 arguments.
```rust
#[allow(clippy::too_many_arguments)]
pub async fn create(
        ctx: &Context,
        node_name: String,
        node_dir: PathBuf,
        // Should be passed only when creating fresh node and we want it to get default root Identity
        identity_override: Option<IdentityOverride>,
        skip_defaults: bool,
        enable_credential_checks: bool,
        ac: Option<&AuthoritiesConfig>,
        project_id: Option<Vec<u8>>,
        projects: BTreeMap<String, ProjectLookup>,
        api_transport: (TransportType, TransportMode, String),
        tcp_transport: TcpTransport,
) -> Result<Self>
```

## Proposed Changes
This PR is meant to fix #3527 
- Extract `NodeManager::create` arguments into three new `Options` structs.
-- `NodeManagerGeneralOptions`, `NodeManagerProjectsOptions` and `NodeManagerTransportOptions`

```rust
pub async fn create(
        ctx: &Context,
        general_options: NodeManagerGeneralOptions,
        projects_options: NodeManagerProjectsOptions<'_>,
        transport_options: NodeManagerTransportOptions,
) -> Result<Self>
```

## Checks

- [x] All commits in this Pull Request are [signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) and Verified by Github.
- [x] All commits in this Pull Request follow the Ockam [commit message convention](https://github.com/build-trust/.github/blob/main/CONTRIBUTING.md#commit-messages).
- [x] I accept the Ockam Community [Code of Conduct](https://github.com/build-trust/.github/blob/main/CODE_OF_CONDUCT.md).
- [x] I have accepted the Ockam [Contributor License Agreement](https://github.com/build-trust/ockam-contributors/blob/main/CLA.md) by adding my Git/Github details in a row at the end of the [CONTRIBUTORS.csv](https://github.com/build-trust/ockam-contributors/blob/main/CONTRIBUTORS.csv) file in a separate pull request to the [build-trust/ockam-contributors](https://github.com/build-trust/ockam-contributors) repository.

### CODE REVIEW

**Comment by vncsalencar** [comment]
Hey @mrinalwadhwa, this is my *work in progress* draft PR.
First of all, I'll rename commits and sign so they follow convention before marking as ready for review.
But I'd like to ask you for some input on the following issues:
- I've tried grouping options based on what I've read and understood from the places where `NodeManager::create` is used. Do you agree with the `...Options` structs I've created or would there be a better suggestion?
- I haven't created `::new` or `::default` methods for these new `structs`, so that's why the struct properties are set to `pub` now. I haven't found any convention regarding that on the project. So, would you like for those methods to be created?

The last one is more because of my inexperience with Rust in big projects. 
- How can I run the `tests` mod inside `service.rs`? It does not seem to be run by using `cargo nextest run` or `cargo test` and I haven't found a way of running it anywhere?

**Comment by SanjoDeundiak** [comment]
Hey @vncsalencar , tnx for your contribution. Answering your questions:
1. I think `Options` suffix makes sense here, I have no objections. Also, it's easy to rename later if needed
2. We have few approaches in the codebase, but I would suggest following:
    - Implement or derive `Default` trait where it makes sense
    - Make fields private with getters
    - Create a constructor (`::new`). It may take all of the fields as arguments if there aren't too many. If makes sense, some of them (that are rarely used/optional) may be not included in the constructor, must have setters then
3. `cargo test` should run enough tests in your case. Also `cargo fmt` and `cargo clippy` may be useful

Let me know if you need anything else

File: implementations/rust/ockam/ockam_api/src/nodes/service.rs [hunk 1]
```diff
@@ -145,36 +145,48 @@ impl NodeManager {
     }
 }
 
+pub struct NodeManagerGeneralOptions<'a> {
+    pub node_name: String,
+    pub node_dir: PathBuf,
+    pub skip_defaults: bool,
+    pub enable_credential_checks: bool,
+    pub ac: Option<&'a AuthoritiesConfig>,
```
Comment (SanjoDeundiak) [inline implementations/rust/ockam/ockam_api/src/nodes/service.rs]: I would move that to `NodeManagerProjectsOptions`

Comment (vncsalencar) [reply implementations/rust/ockam/ockam_api/src/nodes/service.rs]: Sorry, which arguments would you move? `ac` or all of them?

Comment (SanjoDeundiak) [reply implementations/rust/ockam/ockam_api/src/nodes/service.rs]: `ac` only

**Comment by SanjoDeundiak** [comment]
@mergifyio rebase

**Comment by SanjoDeundiak** [comment]
@vncsalencar Thank you for spending time on this, before we can merge please

 Accept the Ockam [Contributor License Agreement](https://github.com/build-trust/ockam-contributors/blob/main/CLA.md) by adding your Git/Github details in a row at the end of the [CONTRIBUTORS.csv](https://github.com/build-trust/ockam-contributors/blob/main/CONTRIBUTORS.csv) file in a separate pull request to the [build-trust/ockam-contributors](https://github.com/build-trust/ockam-contributors) repository.
The easiest way to do this is to [edit the CONTRIBUTORS.csv](https://github.com/build-trust/ockam-contributors/edit/main/CONTRIBUTORS.csv) file in the github web UI and create a PR, this will make the commit as verified.

Also, could you please look at why CI fails? Probably needs `cargo fmt` and [commit message convention](https://github.com/build-trust/.github/blob/main/CONTRIBUTING.md#commit-messages) fix

**Comment by vncsalencar** [comment]
@SanjoDeundiak ran `cargo fmt` and pushed it now. Re-requesting permission for the CI to run.

**Comment by vncsalencar** [comment]
Sorry, had trouble with `gpg` on my side, I rebased and signed + verified the commits.

**Comment by SanjoDeundiak** [comment]
@mergifyio rebase

**Review by SanjoDeundiak — APPROVED** [review]

**Pull request merged**